internal/pkg/cli/deploy/workload.go

```diff
@@ -367,24 +392,86 @@ func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) err
 	if err != nil {
 		return fmt.Errorf("login to image repository: %w", err)
 	}
+
+	var mu sync.Mutex
 	out.ImageDigests = make(map[string]ContainerImageIdentifier, len(buildArgsPerContainer))
+
+	// counter for indexing the labeled output buffers.
+	bufferIdx := 0
+	// An array of buffers, one for each container image build and push output.
+	labeledBuffers := make([]*syncbuffer.LabeledSyncBuffer, len(buildArgsPerContainer))
+
+	// create a context and an error group.
+	g, ctx := errgroup.WithContext(context.Background())
+
+	// create a new cursor and hide it.
+	cursor := cursor.New()
+	cursor.Hide()
+
 	for name, buildArgs := range buildArgsPerContainer {
+		// create a copy of loop variables to avoid data race.
+		name := name
+		buildArgs := buildArgs
+
+		// create a pipe for streaming the build and push output.
+		pr, pw := io.Pipe()
```

Comment:
nit: can we declare this right before the `g.Go` calls? (moving the variable closer to where it's used)